### PR TITLE
Resolve issue where a spawn process' slab fails

### PR DESCRIPTION
**Pull request by jnwatson**
Previously, if a main process' slab mapsize increases, any existing spawn processes (which have the same slabs open readonly) would get a lmdb.MapResizedError when calling slab.initdb.

Now, initdb correctly handles that case.

### CODE REVIEW

File: synapse/tests/test_lib_lmdbslab.py [hunk 1]
```diff
@@ -890,6 +892,40 @@ async def test_slab_mapfull_drop(self):
...
+            async with await s_lmdbslab.Slab.anit(path, map_size=32000) as slab:
+                pass
+            async with await s_lmdbslab.Slab.anit(path, map_size=32000, readonly=True) as slab:
+
+                proc = mpctx.Process(target=self.make_slab, args=(path,))
+                proc.start()
+                proc.join(10)
+                self.nn(proc.exitcode)
+                slab.initdb('foo')
+                self.true(True)
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_lmdbslab.py:927]: is this just testing that the code doesn't go boom?

Comment (invisig0th) [reply synapse/tests/test_lib_lmdbslab.py:927]: Patch coverage of 100% seems to indicate that it covers triggering the actual resize, right?

Comment (jnwatson) [reply synapse/tests/test_lib_lmdbslab.py:927]: This tests that the previous line *didn't* hit an exception (which it did before the change).

Comment (jnwatson) [reply synapse/tests/test_lib_lmdbslab.py:927]: (It is redundant, but I know you like the explicitness)

Comment (vEpiphyte) [reply synapse/tests/test_lib_lmdbslab.py:927]: :+1:

**Review by vEpiphyte — APPROVED** [review]
Approved once CI gremlins are done doing their magick..

**Comment by vEpiphyte** [comment]
Added patches to increase the timeout for the Spawn stormpkg test and a patch from Neomorph for a teardown race on the Slab when we delete the environment.

**Pull request merged**